diff --git a/node_modules/astro/dist/core/build/static-build.js b/node_modules/astro/dist/core/build/static-build.js
index a6b608b..27c1c7d 100644
--- a/node_modules/astro/dist/core/build/static-build.js
+++ b/node_modules/astro/dist/core/build/static-build.js
@@ -189,7 +189,8 @@ ${bgGreen(black(" building client "))}`);
     root: viteConfig.root,
     envPrefix: "PUBLIC_",
     server: viteConfig.server,
-    base: astroConfig.base
+    base: astroConfig.base,
+    resolve: viteConfig.resolve
   };
   await runHookBuildSetup({
     config: astroConfig,
